catalog/ui/catalog-ui-search/src/main/webapp/component/sort-item/sort-item.view.js

```diff
@@ -36,45 +38,141 @@ define([
+        onBeforeShow: function () {
+            var sortAttributes = metacardDefinitions.sortedMetacardTypes.filter(function (type) {
                 return !properties.isHidden(type.id);
-            }).filter(function(type){
+            }).filter(function (type) {
                 return !metacardDefinitions.isHiddenTypeExceptThumbnail(type.id);
-            }).filter(function(type) {
+            }).filter(function (type) {
                 return blacklist.indexOf(type.id) === -1;
-            }).map(function(metacardType) {
+            }).map(function (metacardType) {
                 return {
                     label: metacardType.alias || metacardType.id,
                     value: metacardType.id
                 };
             });
 
-            this.sortAttribute.show(DropdownView.createSimpleDropdown({
-                hasFiltering: true,
-                list: sortAttributes,
-                defaultSelection: [this.model.get('attribute')]
+            if (this.options.showBestTextOption) {
+                sortAttributes.unshift({
+                    label: 'Best Text Match',
+                    value: 'RELEVANCE'
+                });
+            }
+
+            this.sortAttribute.show(new PropertyView({
+                model: new Property({
+                    enum: sortAttributes,
+
+                    value: [this.model.get('attribute')],
+                    id: 'Sort Attribute',
+                    enumFiltering: true
+                })
             }));
-            this.sortDirection.show(DropdownView.createSimpleDropdown({
-                list: [{
-                    label: 'Ascending',
-                    value: 'ascending'
-                }, {
-                    label: 'Descending',
-                    value: 'descending'
-                }],
-                defaultSelection: [this.model.get('direction')]
+            this.handleAttribute();
+            this.turnOnEditing();
+            this.turnOnLimitedWidth();
+
+            this.sortAttribute.currentView.$el.on('change', function (event) {
+                this.handleAttribute();
+            }.bind(this));
+        },
+        turnOffEditing: function () {
+            this.sortAttribute.currentView.turnOffEditing();
+            this.sortDirection.currentView.turnOffEditing();
+        },
+        turnOnEditing: function () {
+            this.sortAttribute.currentView.turnOnEditing();
+            if (!this.$el.hasClass('is-non-directional-sort')) {
+                this.sortDirection.currentView.turnOnEditing();
+            }
+        },
+        turnOnLimitedWidth: function () {
+            this.sortAttribute.currentView.turnOnLimitedWidth();
+            this.sortDirection.currentView.turnOnLimitedWidth();
+        },
+        handleAttribute: function () {
+            var attribute = this.sortAttribute.currentView.getCurrentValue()[0]; //this.model.get('attribute');
+            var ascendingLabel, descendingLabel;
```

Comment:
I think most of this method could be encapsulated into an attribute sorting model.